### PR TITLE
fix(pserver): use with-blocks when possible

### PR DESCRIPTION
**Pull request by michaelfig**
More use of `with` blocks to keep file descriptor usage low.

Part of the review of #382.

### CODE REVIEW

File: packages/cosmic-swingset/provisioning-server/src/ag_pserver/main.py [hunk 1]
```diff
@@ -88,32 +88,32 @@ class ConfigElement(Element):
...
-            f = open(gr)
-            meta['package_git'] = f.read().strip()
+            with open(gr) as f:
+                meta['package_git'] = f.read().strip()
         else:
-            f = os.popen('git rev-parse --short HEAD')
-            sha = f.read().strip()
-            f = os.popen('git diff --quiet || echo -dirty')
-            meta['package_git'] = sha + f.read().strip()
+            with os.popen('git rev-parse --short HEAD') as f:
```
Comment (warner) [inline packages/cosmic-swingset/provisioning-server/src/ag_pserver/main.py:98]: huh, it hadn't occurred to me that `os.popen` provides a context manager, but I guess it makes sense

**Review by warner — APPROVED** [review]
looks good.. didn't realize there were so many :)

**Comment by dckc** [comment]
While you're in the neighborhood of cleanin up `open()` stuff, any chance you want to tackle the ambient authority?

I started cleaning up `ag_setup_solo/main.py` a while ago so I could understand what it was doing...

https://github.com/dckc/cosmic-swingset/commit/7a85ca833bfc69db3adf073230769bab66838d84

**Comment by michaelfig** [comment]
I'd rather wait on that until we move the provisioning server to a cloud platform (which probably means it will be rewritten/redesigned in Node.js).  That's when I intend to remove the magic wormhole dependency, too.

"Remove ambient authority" is a little more far-reaching than "use with-blocks", which was purely mechanical.  Although I'm sure the exercise would be enlightening, I can't just toss that one off in two minutes.

**Pull request merged**